base/shell.jl

```diff
@@ -4,7 +4,19 @@
 
 const shell_special = "#{}()[]<>|&*?~;"
 
-# strips the end but respects the space when the string ends with "\\ "
+"""
+    rstrip_shell(s::AbstractString)
+
+Strip trailing whitespace from a shell command string, while respecting a trailing backslash followed by a space ("\\ ").
+
+```jldoctest
+julia> Base.rstrip_shell("echo 'Hello World' \\ ")
```

Comment:
Some escaping issues here:
https://buildkite.com/julialang/julia-master/builds/31167#018c647a-7353-4bf8-8347-ac5bdc21622e/746-800

Reply:
That is expected, since this is first parsed as a string, then parsed as syntax, so you need to double all `\\` or use `@doc raw""" ... """` to bypass that first unescape pass